eng/pipelines/templates/steps/generate-project-list-and-cache-maven-repository.yml

```diff
@@ -46,6 +51,19 @@ steps:
     ARTIFACTSJSON: $(ArtifactsJson)
     ADDITIONALMODULESJSON: $(AdditionalModulesJson)
     PACKAGEINFODIR: $(Build.ArtifactStagingDirectory)/PackageInfo
+  condition: eq(variables['ArtifactPackageNames'], '')
+
+- task: PowerShell@2
+  displayName: Initialize project list variable for PR test run
+  inputs:
+    pwsh: true
+    filePath: eng/pipelines/scripts/generate-project-list.ps1
+  env:
+    ARTIFACTSJSON: $(ArtifactsJson)
+    ADDITIONALMODULESJSON: $(AdditionalModulesJson)
+    PACKAGEINFODIR: $(Build.ArtifactStagingDirectory)/PackageInfo
+    ARTIFACTPACKAGENAMES: $(ArtifactPackageNames)
```

Comment:
Do you want to fork these steps? You can just do `$env:ArtifactPackageNames` if you want the property to be null if not set.

Reply:
Yes, I want to fork these. The `$env:ArtifactPackageNames` didn't actually work, I had to set it the way I'd set it above for it to work.